### PR TITLE
add twistedchecker ratcheting using twistedchecker --diff

### PR DESCRIPTION
**Pull request by glyph**
- check with twistedchecker
- add a ratchet
- split separate test things into test matrix

### CODE REVIEW

**Comment by glyph** [comment]
This branch doesn't actually use tox, since tox tends to complain about out-of-environment tooling, and this really wants to run `git`.  However, it does adjust `.travis.yml` to look *like* it uses `tox`, so that we can migrate in that direction more easily as we add more interesting testing environments.

**Comment by glyph** [comment]
@chellygel - thoughts?  the extra documentation in `amputil` was mainly to test eliminating some warnings, but it might also be interesting to you in its own right.

File: .travis.yml [hunk 1]
```diff
@@ -13,15 +13,19 @@ install:
   - pip install pyflakes
   - pip install codecov
   - pip install pretend
+  - pip install git+https://github.com/twisted/twistedchecker@9cec9f5b40d1b5cb4d6dcda26ec0ba6e05de4b62#egg=twistedchecker
```
Comment (chellygel) [inline .travis.yml]: is this pegging a specific version? just curious why. little confused on this.

File: vertex/amputil.py [hunk 2]
```diff
@@ -1,51 +1,126 @@
...
 
-    loader = CertificateRequest
 
-class Cert(_BinaryLoadable):
+
+@_argumentForLoader(CertificateRequest)
+class CertReq(Argument):
     """
-    Amp Argument that serializes and deserializes L{Certificate}s 
+    AMP Argument that serializes and deserializes L{CertificateRequest}s.
```
Comment (chellygel) [inline vertex/amputil.py]: whoooopsie. 😨 my mistake!

**Comment by chellygel** [comment]
looks great, but curious about that pip install, but it is just something new to me, can explain that outside of this if you want. :)

File: .travis.yml (hunk 1 above)
Comment (glyph) [inline .travis.yml]: Yes. Specifically, that commit corresponds to this PR - https://github.com/twisted/twistedchecker/pull/114/files - which is necessary for twistedchecker to parse all of vertex without barfing.

File: vertex/amputil.py (hunk 2 above)
Comment (glyph) [inline vertex/amputil.py]: Under Twisted's usual policy of responsibility, it's the reviewer's fault, not the author's, so - no, *my* bad :)

File: .travis.yml (hunk 1 above)
Comment (glyph) [inline .travis.yml]: Also, this should totally be removed once a release of twistedchecker exists with this fix in it.

**Comment by glyph** [comment]
OK! I will also take that as a passing review - assuming my explanation of the pip install thing was OK.  But, for future reference: one of the hallmarks of the twisted review process is a discrete handoff where you say "please address the comments and land" or "please address the comments and resubmit for review".

**Pull request merged**